src/evo/providertx.cpp

```diff
@@ -36,6 +36,11 @@ bool CProRegTx::IsTriviallyValid(bool is_basic_scheme_active, TxValidationState&
     if (!scriptPayout.IsPayToPublicKeyHash() && !scriptPayout.IsPayToScriptHash()) {
         return state.Invalid(TxValidationResult::TX_BAD_SPECIAL, "bad-protx-payee");
     }
+    for (const NetInfoEntry& entry : netInfo.GetEntries()) {
```

Comment:
why `CProRegTx` doesn't validate if `netInfo.IsEmpty()`?

Reply:
Because a ProRegTx is allowed to submit no address (and update it later with a ProUpServTx) but a ProUpServTx must always include an address, see below.

ProRegTx:

https://github.com/dashpay/dash/blob/9cb43c2958f88e7b8d63268dd9252ee86973183a/src/rpc/evo.cpp#L371-L386

https://github.com/dashpay/dash/blob/9cb43c2958f88e7b8d63268dd9252ee86973183a/src/rpc/evo.cpp#L77-L80

ProServUpTx:

https://github.com/dashpay/dash/blob/9cb43c2958f88e7b8d63268dd9252ee86973183a/src/rpc/evo.cpp#L82-L85

https://github.com/dashpay/dash/blob/9cb43c2958f88e7b8d63268dd9252ee86973183a/src/rpc/evo.cpp#L894-L903